### PR TITLE
Bazel build rules moving into Angular repo

### PR DESCRIPTION
**Pull request by alexeagle**
See design: https://goo.gl/rAeYWx

### CODE REVIEW

**Review by chuckjaz — APPROVED** [review]
Assuming lint error is addressed.

**Comment by alexeagle** [comment]
@chuckjaz the scope of this PR has increased, now includes an example bazel app as an integration test, as well as the real ng_module implementation, you should probably take another look.

@IgorMinar this needs your approve for the new package.

File: integration/bazel/src/shared/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (tbosch) [inline integration/bazel/src/shared/BUILD.bazel]: why are these under `shared`?

**Review by tbosch — APPROVED** [review]

File: integration/bazel/src/shared/BUILD.bazel (hunk 1 above)
Comment (alexeagle) [inline integration/bazel/src/shared/BUILD.bazel]: This was following the example from the sass rules
https://github.com/bazelbuild/rules_sass#basic-example

I'd prefer to lay these out following the angular style guide, but I didn't spend the time to learn it...

File: integration/bazel/src/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,10 @@
+load("@angular//:index.bzl", "ng_module")
+
+exports_files(["tsconfig.json"])
```
Comment (IgorMinar) [inline integration/bazel/src/BUILD.bazel]: where is this rule coming from? what does it do?

File: integration/bazel/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "name": "angular-bazel",
+  "description": "example and integration test for building Angular apps with Bazel",
+  "version": "0.0.0",
+  "license": "MIT",
+  "dependencies": {
+    "@angular/animations": "file:../../dist/packages-dist/animations",
+    "@angular/bazel": "file:../../dist/packages-dist/bazel",
+    "@angular/common": "file:../../dist/packages-dist/common",
+    "@angular/compiler": "file:../../dist/packages-dist/compiler",
+    "@angular/compiler-cli": "file:../../dist/packages-dist/compiler-cli",
```
Comment (alexeagle) [inline integration/bazel/package.json]: done

File: integration/bazel/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    "@angular/core": "file:../../dist/packages-dist/core",
+    "@angular/platform-browser": "file:../../dist/packages-dist/platform-browser",
+    "rxjs": "5.3.1",
+    "typescript": "~2.3.1",
```
Comment (alexeagle) [inline integration/bazel/package.json]: done (and `@angular/bazel` too)

File: integration/bazel/src/BUILD.bazel (hunk 2 above)
Comment (alexeagle) [inline integration/bazel/src/BUILD.bazel]: added a comment (it's required so subpackages can reference the tsconfig label)

File: packages/bazel/package.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "@angular/bazel",
+  "version": "0.0.0-PLACEHOLDER",
```
Comment (IgorMinar) [inline packages/bazel/package.json]: resolved in person

File: packages/bazel/src/ngc-wrapped/index.ts [hunk 6]
```diff
@@ -0,0 +1,192 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+// TODO(chuckj): Remove the requirment for a fake 'reflect` implementation from
+// the compiler
+import 'reflect-metadata'; // from //third_party/javascript/node_modules/reflect_decorators:ts
```
Comment (IgorMinar) [inline packages/bazel/src/ngc-wrapped/index.ts]: I believe that this polyfill is no longer needed at HEAD

**Review by IgorMinar — APPROVED** [review]

File: npm-shrinkwrap.clean.json [hunk 7]
```diff
@@ -2329,383 +2340,6 @@
     "fs.realpath": {
       "version": "1.0.0"
     },
-    "fsevents": {
```
Comment (IgorMinar) [inline npm-shrinkwrap.clean.json]: why was `fsevents` removed when what you wanted to remove was `protobufjs`?

**Review by IgorMinar — APPROVED** [review]
the rest looks reasonable. :)

File: npm-shrinkwrap.clean.json (hunk 7 above)
Comment (alexeagle) [inline npm-shrinkwrap.clean.json]: fixed with npm 3.10.7

**Pull request closed without merging**